src/info/index_error.c

```diff
@@ -26,40 +27,41 @@ char* const IndexingErrorKey_String = "last indexing error key";
 char* const IndexingErrorTime_String = "last indexing error time";
 char* const BackgroundIndexingOOMfailure_String = "background indexing status";
 char* const outOfMemoryFailure = "OOM failure";
-RedisModuleString* NA_rstr = NULL;
 
-static void initDefaultKey() {
-    NA_rstr = RedisModule_CreateString(RSDummyContext, NA, strlen(NA));
-    RedisModule_TrimStringAllocation(NA_rstr);
-}
+/*
+* Uninitialized error -> NULL
+* No error -> NA
+* Destroyed error object -> 0xDEADBEEF
+*/
 
 IndexError IndexError_Init() {
-    if (!NA_rstr) initDefaultKey();
     IndexError error = {0}; // Initialize all fields to 0.
     error.last_error_without_user_data = NA;  // Last error message set to NA.
     error.last_error_with_user_data = NA;  // Last error message set to NA.
     // Key of the document that caused the error set to NA.
-    error.key = RedisModule_HoldString(RSDummyContext, NA_rstr);
+    error.key = NULL;
     return error;
 }
 
 static inline void IndexError_ClearLastError(IndexError *error) {
-    if (error->last_error_without_user_data != NA) {
+    RS_ASSERT(error->last_error_without_user_data != DEADBEEF && error->last_error_with_user_data != DEADBEEF);
+    if (error->last_error_without_user_data && error->last_error_without_user_data != NA) {
         rm_free(error->last_error_without_user_data);
     }
-    if (error->last_error_with_user_data != NA) {
+    if (error->last_error_with_user_data && error->last_error_with_user_data != NA) {
         rm_free(error->last_error_with_user_data);
     }
 }
```

Comment:
can we check that error->key != NULL in the begining of the function and skip the NA check ?